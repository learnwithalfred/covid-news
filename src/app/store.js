import { configureStore } from '@reduxjs/toolkit';
import covidReducer from '../features/covid/covidSlice';

const store = configureStore({
  reducer: {
    covid: covidReducer,
  },
});

export default store;
