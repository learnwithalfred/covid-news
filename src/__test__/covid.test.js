import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Covid from '../features/covid/covid';
import store from '../app/store';

describe('Covid should render correctly', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Covid />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('container should have first child', () => {
    const { container } = render(
      <Provider store={store}>
        <Covid />
      </Provider>,
    );
    expect(container.firstChild).toBeTruthy();
  });

  it('Page should have a h1 header', () => {
    const { container } = render(
      <Provider store={store}>
        <Covid />
      </Provider>,
    );
    expect(container.querySelector('h1')).toBeTruthy();
  });
});
