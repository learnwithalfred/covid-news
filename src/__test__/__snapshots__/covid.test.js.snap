// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Covid should render correctly should render correctly 1`] = `
<div>
  <div>
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Covid 19 Tracker
        </a>
        <button
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarSupportedContent"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarSupportedContent"
        >
          <ul
            class="navbar-nav me-auto mb-2 mb-lg-0"
          >
            <li
              class="nav-item dropdown"
            >
              <a
                aria-expanded="false"
                class="nav-link dropdown-toggle"
                data-bs-toggle="dropdown"
                href="/"
                id="navbarDropdown"
                role="button"
              >
                Search Continent Data
              </a>
              <ul
                aria-labelledby="navbarDropdown"
                class="dropdown-menu dropdown-menu-dark"
              >
                <li>
                  <button
                    class="dropdown-item"
                    type="button"
                  >
                    Africa
                  </button>
                </li>
                <li>
                  <button
                    class="dropdown-item"
                    type="button"
                  >
                    Asia
                  </button>
                </li>
                <li>
                  <button
                    class="dropdown-item"
                    type="button"
                  >
                    Europe
                  </button>
                </li>
                <li>
                  <button
                    class="dropdown-item"
                    type="button"
                  >
                    North America
                  </button>
                </li>
                <li>
                  <button
                    class="dropdown-item"
                    type="button"
                  >
                    Oceania
                  </button>
                </li>
                <li>
                  <button
                    class="dropdown-item"
                    type="button"
                  >
                    South America
                  </button>
                </li>
              </ul>
            </li>
          </ul>
          <div
            class="d-flex"
          >
            <input
              aria-label="Search"
              class="form-control me-2"
              name="search"
              placeholder="Search"
              type="search"
              value=""
            />
            <button
              class="btn btn-primary me-2"
              type="submit"
            >
              Search
            </button>
          </div>
        </div>
      </div>
    </nav>
    <main
      class="container py-5"
    >
      <h1 />
      <div
        class="row row-cols-1 row-cols-sm-2 row-cols-md-3 g-3"
      >
        <div
          style="margin: 2rem auto;"
        >
          <div
            style="fill: white; height: 100px; width: 100px;"
          >
            <svg
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
  
              <path
                d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
                opacity=".25"
              />
              
  
              <path
                d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
              >
                
    
                <animatetransform
                  attributeName="transform"
                  dur="0.8s"
                  from="0 16 16"
                  repeatCount="indefinite"
                  to="360 16 16"
                  type="rotate"
                />
                
  
              </path>
              

            </svg>
            

          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
